qiskit/visualization/circuit/latex.py

```diff
@@ -416,7 +417,10 @@ def _build_latex_array(self):
                 num_cols_op = 1
                 wire_list = [self._wire_map[qarg] for qarg in node.qargs if qarg in self._qubits]
                 if getattr(op, "condition", None):
-                    self._add_condition(op, wire_list, column)
+                    if isinstance(op.condition, expr.Expr):
+                        warn("ignoring expression condition, which is not supported yet")
```

Comment:
Do  you think `UserWarning` is appropriate here? I guess we want people to see this, but I'm wondering if it'll be too noisy? Especially since it's not necessarily user actionable if they're using classical expressions.

Reply:
It should only get shown once per drawing, so I think it's ok.  It's particularly important to be visible in the LaTeX drawer, because it straight up skips drawing _any_ condition.  The text and MPL drawers both make an attempt to show the data dependencies still.

Reply:
Maybe just remove it for mpl and text then and leave it in the docstring. For latex I agree it's not correct and we should warn.

Reply:
Done in 076ec9cec.